### ISSUE TITLE
Android documentation inaccuracy?

### ISSUE THREAD

**Issue by EmperorEarth** [unimodules/react-native-unimodules#120]
Please note, just like in #119, I haven't used `react-native-unimodules` on production yet. Just experimenting when I came across this issue.

I had to `yarn add -E unimodules-app-loader` to get my app to build. I'm not sure if the documentation is wrong or if I have a certain combination of native packages or something that requires that dependency.

Without adding this dependency to my project, I get the following behavior during build:
`Error`
```
FAILURE: Build failed with an exception.

* Where:
Script 'path\to\Appname\node_modules\@unimodules\core\unimodules-core.gradle' line: 21

* What went wrong:
A problem occurred evaluating project ':unimodules-react-native-adapter'.
> Project with path ':unimodules-app-loader' could not be found in project ':unimodules-react-native-adapter'.
```

<details>
<summary>Environment</summary>

* Windows 7
* react-native 0.61.5
* `react-native-unimodules`: `0.8.0` via commit hash
* bare react-native project
</details>

<details>
<summary>android/settings.gradle</summary>

```
rootProject.name = 'Appname'
apply from: '../node_modules/react-native-unimodules/gradle.groovy'
includeUnimodulesProjects()
apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
include ':app'
```
</details>

<details>
<summary>android/app/build.gradle</summary>

```
apply plugin: "com.android.application"

import com.android.build.OutputFile

project.ext.react = [
    entryFile: "index.js",
    enableHermes: true,  // clean and rebuild if changing
]

apply from: "../../node_modules/react-native/react.gradle"
apply from: "../../node_modules/react-native-code-push/android/codepush.gradle"
apply from: '../../node_modules/react-native-unimodules/gradle.groovy'
apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"

def enableSeparateBuildPerCPUArchitecture = false

def enableProguardInReleaseBuilds = false

def jscFlavor = 'org.webkit:android-jsc-intl:+'

def enableHermes = project.ext.react.get("enableHermes", true);

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "com.appname"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "major.minor.patch"
    }
    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
        release {
            if (project.hasProperty('APPNAME_RELEASE_STORE_FILE')) {
                storeFile file(APPNAME_RELEASE_STORE_FILE)
                storePassword APPNAME_RELEASE_STORE_PASSWORD
                keyAlias APPNAME_RELEASE_KEY_ALIAS
                keyPassword APPNAME_RELEASE_KEY_PASSWORD
            }
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        debug {
            signingConfig signingConfigs.debug
            resValue "string", "CodePushDeploymentKey", "\"placeholder\""
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://facebook.github.io/react-native/docs/signed-apk-android.
            minifyEnabled enableProguardInReleaseBuilds
            debuggable false
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
            resValue "string", "CodePushDeploymentKey", "\"${System.getenv("CODEPUSH_DEPLOYMENT_KEY_ANDROID_INTERNAL")}\""
        }
    }
    applicationVariants.all { variant ->
        variant.outputs.each { output ->\
            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation 'androidx.appcompat:appcompat:1.1.0-rc01'
    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0-alpha02'
    addUnimodulesDependencies()

    if (enableHermes) {
        def hermesPath = "../../node_modules/hermes-engine/android/";
        debugImplementation files(hermesPath + "hermes-debug.aar")
        releaseImplementation files(hermesPath + "hermes-release.aar")
    } else {
        implementation jscFlavor
    }
}

task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
```
</details>

<details>
<summary>android/build.gradle</summary>

```
buildscript {
    ext {
        buildToolsVersion = "28.0.3"
        minSdkVersion = 21
        compileSdkVersion = 28
        targetSdkVersion = 28
    }
    repositories {
        google()
        jcenter()
    }
    dependencies {
        classpath("com.android.tools.build:gradle:3.4.2")
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            url("$rootDir/../node_modules/react-native/android")
        }
        maven {
            url("$rootDir/../node_modules/jsc-android/dist")
        }
        google()
        jcenter()
        maven { url "https://jitpack.io" }
    }
}
```
</details>

<details>
<summary>MainApplication.java</summary>

```
package com.appname;

import com.appname.generated.BasePackageList;

import android.app.Application;

import android.content.Context;
import com.facebook.react.PackageList;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.soloader.SoLoader;

import com.microsoft.codepush.react.CodePush;
import org.unimodules.adapters.react.ModuleRegistryAdapter;
import org.unimodules.adapters.react.ReactModuleRegistryProvider;
import org.unimodules.core.interfaces.SingletonModule;

import java.lang.reflect.InvocationTargetException;
import java.util.Arrays;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {
  private final ReactModuleRegistryProvider mModuleRegistryProvider = new ReactModuleRegistryProvider(new BasePackageList().getPackageList(), Arrays.<SingletonModule>asList());

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
    @Override
    protected String getJSBundleFile(){
      return CodePush.getJSBundleFile();
    }

    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      @SuppressWarnings("UnnecessaryLocalVariable")
      List<ReactPackage> packages = new PackageList(this).getPackages();

      List<ReactPackage> unimodules = Arrays.<ReactPackage>asList(
        new ModuleRegistryAdapter(mModuleRegistryProvider)
      );
      packages.addAll(unimodules);

      return packages;
    }

    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    initializeFlipper(this); // Remove this line if you don't want Flipper enabled
  }

  private static void initializeFlipper(Context context) {
    if (BuildConfig.DEBUG) {
      try {
        Class<?> aClass = Class.forName("com.facebook.flipper.ReactNativeFlipper");
        aClass.getMethod("initializeFlipper", Context.class).invoke(null, context);
      } catch (ClassNotFoundException e) {
        e.printStackTrace();
      } catch (NoSuchMethodException e) {
        e.printStackTrace();
      } catch (IllegalAccessException e) {
        e.printStackTrace();
      } catch (InvocationTargetException e) {
        e.printStackTrace();
      }
    }  }
}
```
</details>

**Comment by sezginriggs** [comment]
I also have the same problem and reported with https://github.com/unimodules/react-native-unimodules/issues/122

**Comment by SConaway** [comment]
I added this project to one of mine. Here are a bunch of diffs. I'm using RN v0.62. Hope they help

https://github.com/SConaway/rnfb-test/commit/c18f6ef60665309d2fcaaa2b2bce6a38056c48de#diff-dc46f9f5dfe204e394fb41395ccd03cf
https://github.com/SConaway/rnfb-test/commit/c18f6ef60665309d2fcaaa2b2bce6a38056c48de#diff-74aeda4cd5fc88b4be36cd1a3acefec6
https://github.com/SConaway/rnfb-test/commit/becd0b1f06c7b63f23a7fe6d0323799900a7ec7d#diff-6d153c6178352bc8f8d07ca6fdd1d5cb

**Comment by wood1986** [comment]
Same there

**Comment by RatebSeirawan** [comment]
@SConaway 

> I added this project to one of mine. Here are a bunch of diffs. I'm using RN v0.62. Hope they help
> 
> [SConaway/rnfb-test@c18f6ef#diff-dc46f9f5dfe204e394fb41395ccd03cf](https://github.com/SConaway/rnfb-test/commit/c18f6ef60665309d2fcaaa2b2bce6a38056c48de#diff-dc46f9f5dfe204e394fb41395ccd03cf)
> [SConaway/rnfb-test@c18f6ef#diff-74aeda4cd5fc88b4be36cd1a3acefec6](https://github.com/SConaway/rnfb-test/commit/c18f6ef60665309d2fcaaa2b2bce6a38056c48de#diff-74aeda4cd5fc88b4be36cd1a3acefec6)
> [SConaway/rnfb-test@becd0b1#diff-6d153c6178352bc8f8d07ca6fdd1d5cb](https://github.com/SConaway/rnfb-test/commit/becd0b1f06c7b63f23a7fe6d0323799900a7ec7d#diff-6d153c6178352bc8f8d07ca6fdd1d5cb)

This made the trick thank a lot :) 
after adding includeUnimodulesProjects()
to settings.gradle

**Comment by EmperorEarth** [comment]
Seems to be fixed with `react-native-unimodules@0.14.x`
